src/main/java/org/opensearch/neuralsearch/search/HitsThresholdChecker.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.search;
+
+import java.util.Locale;
+
+import org.apache.lucene.search.ScoreMode;
+
+/**
+ *  Defines algorithm to allow searches to terminate early
+ */
+public class HitsThresholdChecker {
+    private int hitCount;
+    private final int totalHitsThreshold;
+
+    private HitsThresholdChecker(int totalHitsThreshold) {
+        if (totalHitsThreshold < 0) {
+            throw new IllegalArgumentException(String.format(Locale.ROOT, "totalHitsThreshold must be >= 0, got %d", totalHitsThreshold));
+        }
+        assert totalHitsThreshold != Integer.MAX_VALUE;
+        this.totalHitsThreshold = totalHitsThreshold;
+    }
+
+    int getHitsThreshold() {
+        return totalHitsThreshold;
+    }
+
+    void incrementHitCount() {
+        ++hitCount;
+    }
+
+    boolean isThresholdReached() {
+        return hitCount > getHitsThreshold();
+    }
+
+    ScoreMode scoreMode() {
+        return ScoreMode.TOP_SCORES;
```

Comment:
Whats the point of this?

Reply:
But this class does not implement Collector interface

Reply:
right, this method called from doc collector https://github.com/martin-gaievski/neural-search/blob/feature/normalization-query-phase-searcher/src/main/java/org/opensearch/neuralsearch/search/HybridTopScoreDocCollector.java#L93, I found that in other doc collectors it's done in such way, e.g. [TopScoreDocCollector](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java#L314)

Reply:
Shouldnt we extend or use https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/HitsThresholdChecker.java?

Reply:
Shouldn't we extend https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/HitsThresholdChecker.java#L23?